BaseLib/ConfigTree-impl.h

```diff
@@ -50,15 +52,37 @@ getConfParam(std::string const& param, T const& default_value) const
     return default_value;
 }
 
-template<typename T>
-boost::optional<T>
-ConfigTree::
-getConfParamOptional(std::string const& param) const
+template <typename T>
+boost::optional<T> ConfigTree::getConfParamOptional(
+    std::string const& param) const
 {
     checkUnique(param);
 
+    return getConfParamOptionalImpl(param, static_cast<T*>(nullptr));
+}
+
+template <typename T>
+boost::optional<T> ConfigTree::getConfParamOptionalImpl(
```

Comment:
can't we simply write `getConfParamOptionalImpl<T>(param);` by removing the second parameter?